### DISCUSSION TITLE
How can I transfer Stuffit Expander to my premade Mac OS image?

### DISCUSSION THREAD

**Discussion by papioara** [BlueSCSI/BlueSCSI-v2#151] [Help] [answered]
Hello everyone,

I have installed my BlueSCSI board internally in my Mac SE. I have downloaded and written an enhanced version of System 7.0.1 from Macintosh Garden on the SD card. It works out of the box, but now I want to customize it by installing additional software on it. I have used (as recommended) the SCSI Toolbox, have created my 'shared' folder and put some software in it in order to move it to my drive. Unfortunately I am caught in a vicious circle now: When you download software from Macintosh Garden and the like, you mostly get Stuffit archives. That means that the most important software on my drive would be Stuffit Expander. On the SE, I cannot open them, because in the premade image there is no Stuffit Expander. If I unstuff them on my Mac before putting them into the 'shared' folder, I often get a folder with several files in it. Unfortunately the Toolbox cannot transfer folders. If I cheat and put Stuffit Expander into 'shared' separately, I get a Toolbox error saying something about dividing by zero and forcing me to restart the system.

Hopefully someone here can explain what is going on here and if there is another workaround. Otherwise I cannot do much with that premade image because there is no way of unstuffing all that software or to transfer the applications into the system.

Any ideas or suggestions would be greatly appreciated

**Comment by erichelgeson** [comment, marked as answer]
You're in the chicken and egg situation of resource forks for the mac. You can not expand or transfer classic mac files with file systems like FAT32/ExFAT/etc - they do not support resource forks, which will leave your files broken.

The solution is to use our "Bootstrap" image - it contains many popular compression tools pre-installed/expanded so you can just use them. Then you can use the BlueSCSI Toolbox app to transfer the .sit or .bin files to the classic mac, and expand.

https://macintoshgarden.org/apps/piscsi-rascsi - grab the RaSCSI_v3_Bootstrap.hda_.zip download.

Note you'll also run into the Stuffit 5 problem. SE is a 68000 CPU so can not decompress Stuffit 5 archives. For this you'll need to use an emulator - I suggest just using https://infinitemac.org/ in your web browser - drag the file into the browser to upload to the emulator, then expand it there and put it on the "Saved HD". You can export that drive as a .hda. 

It's always a dance with these old macs and file systems and file types, hopefully that's enough to get you going.

**Comment by papioara** [comment]
Thank you for your advice - I'll try it out and see what happens, and of course I will report...

**Comment by papioara** [comment]
Another idea: would it help to format the SD card as HFS instead of FAT?

**Reply by erichelgeson** [reply]
The libraries the Pico uses does not understand HFS, nor does any modern OS. While there is a way to /kind of/ do this with raw mode, I don't recommend it as you loose almost all other features of BlueSCSI.

**Comment by papioara** [comment]
Thank you again. I have looked at the Macintosh Garden page, but there remain questions for me.

1. The two images - the OS and the Bootstrap one: do they go on the SD card? Or onto some directory on my (Mac or Linux) modern computer?
2. If I do the two command line commands: this does obviously have to take place on my modern computer, but except for mounting the images: what is the purpose?

It seems that I miss something...

**Reply by erichelgeson** [reply]
1) Yes, they are drive images for Pi/BlueSCSI - place the Bootstrap one on your SD card as, say `HD2 - Bootstrap.hda` (see the [Usage](https://github.com/blueSCSI/blueSCSI-v2/wiki/Usage) page for naming conventions. 

2) What command line commands?

**Comment by papioara** [comment]
OK, I see. 
On the Macintosh Garden page it says this here:

Usage:
Mount a compatible OS with your machine as well as the Bootstrap image:

scsictl -i 1 -c attach -t hd -f /home/pi/images/RaSCSI-Boot-7.0.1.hda
scsictl -i 2 -c attach -t hd -f /home/pi/images/RaSCSI-BootstrapV3.hda

**Reply by erichelgeson** [reply]
Those commands are for Ra/PiSCSI, you just need the file on your SD card and boot.

**Comment by papioara** [comment]
Allright - I will try that and see what happens. I'll get back.

**Comment by papioara** [comment]
Just downloaded them . Do I have to rename them according to the naming convention or can I leave the file names as they are? Trying to avoid some trial and error...

**Comment by papioara** [comment]
I have tried both possibilities with the files - with and without SCSI ID. Only with SCSI ID lets the SE boot. But then a message appears on the screen saying that the desktop file cannot be built, and that I should try unlocking the disk. But how on earth do I "unlock" that disk?

**Reply by erichelgeson** [reply]
Please see the Usage page, it indicates the file naming convention. You should be using the bootstrap image and the existing system image you had before this. Which image is complaining it's locked?

**Comment by papioara** [comment]
Oops, I have used the wrong image. Returning to the previous working one ond the Bootstrap image named properly works. Great tool collection on the Bootstrap volume, but unfortunately still no good news  I tried a .sit file on Stufft Lite 3.6, because it said in the description that it was a Stuffit Lite 3.6 archive. But it seems that Stuffit Lite does not recognize my archive as a Stuffit archive. The only difference now is: before actually being able to use Stuffit Lite (it launches when you drop the file on it), the error message was that the program which had created that file was not available on this computer.

And now: This does not appear to be a Stuffit file... And antother question: does it really matter which OS you use for downloading and copying files to the SD card? I have been using an old Mac with OSX Tiger and also a Linux machine. 

What is wrong here, and how can it be fixed?

**Reply by erichelgeson** [reply]
Can you provide a link to the actual file you're trying to use? It may be a Stuffit 5.5 file which you will need to repack into something the SE can decompress. Try Stuffit expander 4(thats the latest one you can use, it is also on the image)

It does not matter which OS you use to download files - you will never have the resource fork (as http download cant transfer the resource fork to begin with). The file will just "look" like a plain file - and you'll need to drag and drop it onto stuffit expander to expand.

**Comment by papioara** [comment]
Here you go: [https://www.macintoshrepository.org/9086-drachen](url)

Just a little b/w game and I wanted to use it as some kind of a test dummy...

**Reply by erichelgeson** [reply]
Indeed this is a Stuffit 5 archive. There is a Stuffit 1.5.1 archive of this game (which you will be able to decompress) here as Download #2 - http://macintoshgarden.org/games/dragon-0

**Comment by papioara** [comment]
I'll try that one, thank you. But how can you tell it's a Stuffit.5 one?

Am 26. Mai 2024 17:43:56 MESZ schrieb Eric Helgeson ***@***.***>:
>Indeed this is a Stuffit 5 archive. There is a Stuffit 1.5.1 archive of this game (which you will be able to decompress) here as Download #2 - http://macintoshgarden.org/games/dragon-0
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/BlueSCSI/BlueSCSI-v2/discussions/151#discussioncomment-9562292
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- Von /e/OS E-Mail gesendet.

**Reply by erichelgeson** [reply]
Open it up in a hex editor and look at the file header - not a trivial thing to do unfortunately.

**Comment by papioara** [comment]
I'll do so. But it's funny my Ultimate 7.1 lies to me about the version. So then every time I want to move a .sit file to my 'shared' folder, I need to check with which Stuffit version it has been created. Tedious, but what can you do... Can you recommend a good hex editor for either Mac or Linux?

**Comment by papioara** [comment]
Thanks for the pointer to the hex editors. I have installed hexcurse and I have compared the two files. I can confirm your findings And the header of such a file is not hard to find - it's at the top and a bit cryptic- as the name header suggests. hexcurse has a nice split window.
I will try it out tomorrow and keep my fingers crossed...

**Reply by erichelgeson** [reply]
I usually use just `hexdump -C file.sit | head -n1` cli or [ImHex](https://github.com/WerWolv/ImHex). No header is not hard to find if you're comfortable with hex editors, though not everyone is at the same level, didn't want to assume.
